metadata-ingestion/docs/sources/iceberg/iceberg.md

```diff
@@ -1,4 +1,175 @@
+configuration option - refer to `pyiceberg` documentation on details.
+
+### Glue catalog + S3 warehouse
+
+The minimal config for connecting to such
+
+```yaml
+source:
+  type: "iceberg"
+  config:
+    env: dev
+    catalog:
+      my_catalog:
+        type: "glue"
+        s3.region: "us-west-2"
+        region_name: "us-west-2"
+```
+
+Where `us-west-2` is the region from which you want to ingest. Above will work assuming your pod or environment in which
+you run your datahub CLI is already authenticated to AWS and has proper permissions granted (see below). If you need
+to specify secrets directly, then use below as the template:
+
+```yaml
+source:
+  type: "iceberg"
+  config:
+    env: dev
+    catalog:
+      demo:
+        type: "glue"
+        s3.region: "us-west-2"
+        s3.access-key-id: "${AWS_ACCESS_KEY_ID}"
+        s3.secret-access-key: "${AWS_SECRET_ACCESS_KEY}"
+        s3.session-token: "${AWS_SESSION_TOKEN}"
+
+        aws_access_key_id: "${AWS_ACCESS_KEY_ID}"
+        aws_secret_access_key: "${AWS_SECRET_ACCESS_KEY}"
+        aws_session_token: "${AWS_SESSION_TOKEN}"
+        region_name: "us-west-2"
+```
+
+Above example uses references to fill credentials (either from Secrets defined in Managed Ingestion or environmental variables).
+It is possible (but not recommended due to security concerns) to provide those values in plaintext, directly in the recipe.
+
+#### Glue and S3 permissions required
+
+The role used by the ingestor for ingesting metadata from Glue Iceberg Catalog and S3 warehouse is:
+
+```json
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": ["glue:GetDatabases", "glue:GetTables", "glue:GetTable"],
+      "Resource": "*"
+    },
+    {
+      "Effect": "Allow",
+      "Action": ["s3:GetObject", "s3:ListBucket", "s3:GetObjectVersion"],
+      "Resource": [
+        "arn:aws:s3:::<bucket used by the warehouse>",
+        "arn:aws:s3:::<bucket used by the warehouse>/*"
+      ]
+    }
+  ]
+}
+```
+
+### Iceberg REST Catalog + MinIO
+
+Below assumes MinIO defines authentication specified with `s3.*` prefix. Note specification of `s3.endpoint`, assuming
+MinIO listens on port `9000` at `minio-host`. The `uri` parameter points at IRC endpoint (in this case `iceberg-catalog:8181`).
+
+```yaml
+source:
+  type: "iceberg"
+  config:
+    env: dev
+    catalog:
+      demo:
+        type: "rest"
+        uri: "http://iceberg-catalog:8181"
+        s3.access-key-id: "${AWS_ACCESS_KEY_ID}"
+        s3.secret-access-key: "${AWS_SECRET_ACCESS_KEY}"
+        s3.region: "eu-east-1"
+        s3.endpoint: "http://minio-host:9000"
+```
+
+### Iceberg REST Catalog (with authentication) + S3
+
+This example assume IRC requires token authentication (via `Authorization` header). There are more options available,
+see https://py.iceberg.apache.org/configuration/#rest-catalog for details. Moreover, the assumption here is that environment
```

Comment:
that environment -> that the environment in which in the ingestion will run in (i.e. pod) is already...